src/models/PolicyList.ts

```diff
@@ -607,3 +624,177 @@ export class Revision {
+            // Note that we bailout *after* the call to `joinRoom`, in case a user
+            // calls `watchList` in an attempt to repair something that was broken,
+            // e.g. a Mjölnir who could not join the room because of alias resolution
+            // or server being down, etc.
+            return null;
+        }
+
+        const list = await this.addPolicyList(roomId, roomRef);
+
+        await this.storeWatchedPolicyLists();
+
+        await this.warnAboutUnprotectedPolicyListRoom(roomId);
+
+        return list;
+    }
+
+    public async unwatchList(roomRef: string): Promise<PolicyList | null> {
+        const permalink = Permalinks.parseUrl(roomRef);
+        if (!permalink.roomIdOrAlias) return null;
+
+        const roomId = await this.mjolnir.client.resolveRoom(permalink.roomIdOrAlias);
+        this.failedStartupWatchListRefs.delete(roomRef);
+        const list = this.policyLists.find(b => b.roomId === roomId) || null;
+        if (list) {
+            this.policyLists.splice(this.policyLists.indexOf(list), 1);
+            this.mjolnir.ruleServer?.unwatch(list);
+            this.mjolnir.protectedRoomsTracker.unwatchList(list);
+        }
+
+        await this.storeWatchedPolicyLists();
+        return list;
+    }
+
+    /**
+     * Load the watched policy lists from account data, only used when Mjolnir is initialized.
+     */
+    public async init() {
+        this.policyLists = [];
+        const joinedRooms = await this.mjolnir.client.getJoinedRooms();
+
+        let watchedListsEvent: { references?: string[] } | null = null;
+        try {
+            watchedListsEvent = await this.mjolnir.client.getAccountData(WATCHED_LISTS_EVENT_TYPE);
+        } catch (e) {
+            if (e.statusCode === 404) {
+                LogService.warn('Mjolnir', "Couldn't find account data for Mjolnir's watched lists, assuming first start.", extractRequestError(e));
+            } else {
+                throw e;
+            }
+        }
+
+        for (const roomRef of (watchedListsEvent?.references || [])) {
+            const permalink = Permalinks.parseUrl(roomRef);
+            if (!permalink.roomIdOrAlias) continue;
+
+            let roomId;
+            try {
+                roomId = await this.mjolnir.client.resolveRoom(permalink.roomIdOrAlias);
+            } catch (ex) {
+                // Let's not fail startup because of a problem resolving a room id or an alias.
+                LogService.warn('Mjolnir', 'Could not resolve policy list room, skipping for this run', permalink.roomIdOrAlias)
+                await this.mjolnir.managementRoomOutput.logMessage(LogLevel.WARN, "Mjolnir", `Room ${permalink.roomIdOrAlias} could **not** be resolved, perhaps a server is down? Skipping this room. If this is a recurring problem, please consider removing this room.`);
+                this.failedStartupWatchListRefs.add(roomRef);
+                continue;
```

Comment:
This needs changing to as per https://github.com/matrix-org/mjolnir/issues/404
We should follow through and add the migration step.

Reply:
Do we also need a generic utility for account data and migrating the data format?

Reply:
we made one

Reply:
`MatrixDataManager`